schema/handler.go

```diff
@@ -77,34 +80,140 @@ type getJsonSchema struct {
+		return
+	}
+}
+
+// Raw JSON Schema map
+//
+// swagger:model jsonSchemas
+// nolint:deadcode,unused
+type jsonSchemas map[string]json.RawMessage
+
+// nolint:deadcode,unused
+// swagger:parameters getJsonSchemas
+type getJsonSchemas struct {
+	// Items per Page
+	//
+	// This is the number of items per page.
+	//
+	// required: false
+	// in: query
+	// default: 100
+	// min: 1
+	// max: 500
+	PerPage int `json:"per_page"`
+
+	// Pagination Page
+	//
+	// required: false
+	// in: query
+	// default: 0
+	// min: 0
+	Page int `json:"page"`
+}
+
+// swagger:route GET /schemas v0alpha1 getJsonSchemas
+//
+// Get all JSON Schemas
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: jsonSchemas
+//       500: jsonError
+func (h *Handler) getAll(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	page, itemsPerPage := x.ParsePagination(r)
+
+	ids, err := h.r.IdentityTraitsSchemas(r.Context()).GetIDs()
+	if err != nil {
+		h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrNotFound.WithDebugf("%+v", err)))
+		return
+	}
+
+	total := int64(len(ids))
+	skip := page * itemsPerPage
+	if skip > len(ids) {
+		skip = len(ids)
+	}
+	items := skip + itemsPerPage
+	if items > len(ids) {
+		items = len(ids)
+	}
+	ids = ids[skip:items]
+
+	ss := map[string]json.RawMessage{}
+
+	for _, id := range ids {
+		s, err := h.r.IdentityTraitsSchemas(r.Context()).GetByID(id)
 		if err != nil {
-			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
+			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrNotFound.WithDebugf("%+v", err)))
 			return
 		}
-		defer src.Close()
-	} else {
-		resp, err := http.Get(s.URL.String())
+		ls, err := readSchema(s)
 		if err != nil {
 			h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
 			return
 		}
-		defer resp.Body.Close()
-		src = resp.Body
+		ss[id] = ls
 	}
 
+	x.PaginationHeader(w, urlx.AppendPaths(h.r.Config(r.Context()).SelfAdminURL(), fmt.Sprintf("/%s", SchemasPath)), total, page, itemsPerPage)
 	w.Header().Add("Content-Type", "application/json")
-	if _, err := io.Copy(w, src); err != nil {
+
+	enc := json.NewEncoder(w)
+	enc.SetIndent("", "  ")
+	if err := enc.Encode(ss); err != nil {
 		h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("The file for this JSON Schema ID could not be found or opened. This is a configuration issue.").WithDebugf("%+v", err)))
 		return
 	}
 }
+
+func readSchema(schema *Schema) (json.RawMessage, error) {
```

Comment:
Maybe instead of JSON decoding and encoding, you can instead return the `io.ReadCloser` here and then use `io.Copy` as before?